lib/OpenQA/Schema/Result/Jobs.pm

```diff
@@ -1833,6 +1834,31 @@ sub test_resultfile_list {
     return \@filelist_existing;
 }
 
+=head2 investigate
+
+Find pointers for investigation on failures, e.g. what changed vs. a "last
+good" job in the same scenario.
+
+=cut
+sub investigate {
+    my ($self, %args) = @_;
+    my @previous = $self->_previous_scenario_jobs;
+    return {error => 'No previous job in this scenario, cannot provide hints'} unless @previous;
+    my %investigation;
+    return {error => 'No result directory available for current job'} unless $self->result_dir();
+    for my $prev (@previous) {
+        next unless ($prev->result =~ '(passed|softfailed)');
+        $investigation{last_good} = $prev->id;
+        last unless $prev->result_dir;
+        $investigation{diff_to_last_good} = eval {
+            diff(map { catfile($_->result_dir(), 'vars.json') } ($prev, $self));
+        };
```

Comment:
Do you really want to ignore all exceptions? If that's the case i would move the whole `map` out of the eval just to be defensive.

Reply:
done